application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,213 @@
+        User author = event.getJDA().getUserById(args.get(0));
+        createForumPost(event, author);
+
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    public static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX
+                && !titleCompact.toLowerCase(Locale.US).contains("help");
+    }
+
+    public void createForumPost(ModalInteractionEvent event, User originalUser) {
+
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+
+        ForumChannel questionsForum = event.getJDA().getForumChannelById(QUESTIONS_CHANNEL_ID);
+
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+
+        ForumTag javaTag = tags.get(0);
+
+        event.getJDA()
+            .getForumChannelById(QUESTIONS_CHANNEL_ID)
+            .createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
+            .map(ForumPost::getMessage)
+            .flatMap(postSent -> event.reply(postSent.getJumpUrl() + " "
+                    + originalUser.getAsMention() + "'s question has been transferred"))
+            .queue(onSuccess -> dmUser(event, originalUser), onFailure -> logger
+                .warn("Failed to Transfer Question to #questions", onFailure));
+    }
+
+
+
+    public void dmUser(ModalInteractionEvent event, User originalUser) {
+        event.getJDA().getUserById(originalUser.getId()).openPrivateChannel().queue(onSuccess -> {
+            onSuccess.sendMessage(
+                    "Hi, you are receiving this dm because you tried to ask question in wrong chat, please use #questions for asking questions, more info #rules -Together Java")
+                .queue(onNestedSuccess -> deleteOriginalMessage(event, originalMessageId),
+                        onNestedFailure -> {
+                            logger.warn("Failed to open private channel for user of message : "
+                                    + originalMessageId, onNestedFailure);
+                            logger.info("sending message to source channel", onNestedFailure);
+                            sendMessageToSource(event, originalUser);
+                        });
+        }, onFailure -> {
```

Comment:
this is blocking

Reply:
Was it because of "not being able to pull user?" or the nested calls? , How about now?
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/f30f2b80-17ee-4375-9d0b-8f6eec5048b7)

Reply:
its because of nested calls